### PR TITLE
[fix]: [CDS-84500]: Adding log info for null inputs but with default value

### PR DESCRIPTION
**Pull request by rchaudharyy**
Adding log info for null inputs but with default value. This change is only for monitoring the log and verifying all the edge case where it might applicable.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56413)

### CODE REVIEW

**Comment by sahilhindwani-harness** [comment]
*[`970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java` line 105 at r1](https://reviewable.io/reviews/harness/harness-core/56413#-Nl1up0E2oiy-wFA1HW2:-Nl1up0E2oiy-wFA1HW3:btd3cno) ([raw file](https://github.com/harness/harness-core/blob/bf9e6d75b4aef3bb47969c30816f0f40eb02b1ab/970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java#L105)):*
> ```Java
>       String defaultValue = NGRuntimeInputUtils.extractDefaultValueFromNullInput(text);
>       log.info("Text containing null with default is {} and extracted default value is {} test", text, defaultValue);
>       //      return ParameterField.createFieldWithDefaultValue(
> ```

remove this comment

**Comment by sahilhindwani-harness** [comment]
*[`970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java` line 104 at r1](https://reviewable.io/reviews/harness/harness-core/56413#-Nl1v3bV9kdIQCug4wNu:-Nl1v3bV9kdIQCug4wNv:b656pmb) ([raw file](https://github.com/harness/harness-core/blob/bf9e6d75b4aef3bb47969c30816f0f40eb02b1ab/970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java#L104)):*
> ```Java
>     if (EmptyPredicate.isNotEmpty(text) && text.contains("null.")) {
>       String defaultValue = NGRuntimeInputUtils.extractDefaultValueFromNullInput(text);
>       log.info("Text containing null with default is {} and extracted default value is {} test", text, defaultValue);
> ```

make this warn.

**Comment by rchaudharyy** [comment]
trigger smartchecks,

**Comment by rchaudharyy** [comment]
trigger smartchecks

**Comment by rchaudharyy** [comment]
trigger gitleaks

**Comment by rchaudharyy** [comment]
trigger uts

File: 970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java [hunk 1]
```diff
@@ -98,6 +101,11 @@ public ParameterField<?> deserialize(JsonParser p, DeserializationContext ctxt)
 
     InputSetValidator inputSetValidator = getInputSetValidator(text);
 
+    if (EmptyPredicate.isNotEmpty(text) && text.contains("null.")) {
+      String defaultValue = NGRuntimeInputUtils.extractDefaultValueFromNullInput(text);
```
Comment (sahilhindwani-harness) [inline 970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java:105]: Add inside try catch block.

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]

**Review by rchaudharyy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56413)* status: 4 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java` line 105 at r3](https://reviewable.io/reviews/harness/harness-core/56413#gh-1436436407:-Nme-wjr3gK9-xkZaUAj:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d517cc80b30b1dbc3db7e3f542cb2e263b62425a/970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java#L105)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

Add inside try catch block.
</blockquote></details>

Done.

**Comment by rchaudharyy** [comment]
trigger uts

**Comment by rchaudharyy** [comment]
trigger smartchecks

**Comment by rchaudharyy** [comment]
trigger checkstylecodeformat

**Comment by rchaudharyy** [comment]
trigger uts

**Comment by sahilhindwani-harness** [comment]
*[`970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java` line 109 at r7](https://reviewable.io/reviews/harness/harness-core/56413#-NmkEuWSCG5ErDM7b4Kx:-NmkEuWSCG5ErDM7b4Ky:b-cqbwy7) ([raw file](https://github.com/harness/harness-core/blob/f1e3874b75fc73eada5246b0fa8d3aa6ed69f43e/970-ng-commons/src/main/java/io/serializer/jackson/ParameterFieldDeserializer.java#L109)):*
> ```Java
>         log.warn("Text containing null with default is {} and extracted default value is {} test", text, defaultValue);
>       } catch (Exception e) {
>         log.error("Error extracting default value from text '{}'", text, e);
> ```

Please use String.format here otherwise exception won't get logged properly.

**Comment by sahilhindwani-harness** [comment]
*[`970-ng-commons/src/test/java/io/serializer/utils/NGRuntimeInputUtilsTest.java` line 77 at r7](https://reviewable.io/reviews/harness/harness-core/56413#-NmkF3Jm-8er5moRpY2-:-NmkF3Jm-8er5moRpY20:b-l6gvng) ([raw file](https://github.com/harness/harness-core/blob/f1e3874b75fc73eada5246b0fa8d3aa6ed69f43e/970-ng-commons/src/test/java/io/serializer/utils/NGRuntimeInputUtilsTest.java#L77)):*
> ```Java
>     assertThat(NGRuntimeInputUtils.extractDefaultValueFromNullInput(defaultValueString)).isEqualTo("3");
> 
>     String defaultValuesWithAllowedValues = "null.allowedValues(2,5).default(5)";
> ```

can we add a test ofr null.allowedValues?

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 3 files at r4, 1 of 1 files at r5, 2 of 2 files at r7, 3 of 3 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56413)* status: 7 file(s) to review. 1 of 2 approvals obtained.

**Comment by rchaudharyy** [comment]
trigger uts

**Review by sandeshsalunkhegh — APPROVED** [review]
LGTM

**Comment by rchaudharyy** [comment]
trigger review
